### ISSUE TITLE
Iteration 3 - Shoring things up so my students can use it

### ISSUE THREAD

**Issue by JoshCheek** [JoshCheek/what-we-ve-got-here-is-an-error-to-communicate#44]
Amend this issue as necessary.

Goal
-------

Be able to make this the default formatter for [mrspec](https://github.com/JoshCheek/mrspec) so that my students can use it. It needs to have decent messages for all cases, and no inhibiting bugs.

Making it generally usable
-----------------------------------

- [x] Make stories (or link them) for any of these that need them (I don't have time to do it right now, I'm going to release 2, commit this, and then chill with some humans).
- [x] RSpec formatter handles assertions badly https://github.com/JoshCheek/what-we-ve-got-here-is-an-error-to-communicate/issues/42 (high b/c this is our primary use case, and affects our own development)

  ![screenshot](https://cloud.githubusercontent.com/assets/77495/7898952/2eec3486-06ce-11e5-9382-7fb4bf3c8295.png)
- [ ] Look through the list of errors, and see if there are any which are sufficiently handled by falling back to Exception. Close any which are. (Closed #29, #28 )
- [ ] Might be worth closing all the low-priority issues, they're tracked in the backlog, we can always go turn them back on, but might be nice to have a finite list that is easy to glance at and see what's going on.

MRspec integration
--------------------------

- [x] Add this as the default formatter for [mrspec](https://github.com/JoshCheek/mrspec)
- [x] Make sure that it can be overridden if the user explicitly chooses to do so
- [x] Make sure that Minitest assertions look good in it (ie RSpec assertions currently look terrible, b/c it doesn't realize what they are, and tries to stick the long formatted message in that first line)
- [x] Once all of this is done, release mrspec with the dependency in place (will need to bump the minor, b/c it adds a dep, so is a breaking change -- mrspec is still pre 1.0). [Done, it is in version 0.2.0](https://rubygems.org/gems/mrspec).

Descoped
--------------

- [ ] [More confident inheritance](https://github.com/JoshCheek/what-we-ve-got-here-is-an-error-to-communicate/issues/46) Played around with a few [ideas](https://gist.github.com/JoshCheek/74d5fa8a7cf341c5b49f), didn't find anything I liked >.< effit, this will just happen at some point if it really matters :P
- [ ] Figure out what to do about exception hierarchies https://github.com/JoshCheek/what-we-ve-got-here-is-an-error-to-communicate/issues/39 We may not actually need this, but I know Minitest's assertions all inherit from one assertion class. So we may have to figure it out. If not, then descope this.

**Comment by JoshCheek** [comment]
I'm going to close this, I don't find anything particularly compelling among the ones that aren't done.
